camel/agents/chat_agent.py

```diff
@@ -431,10 +434,25 @@ def step(
             for base_message_item in output_messages:
                 base_message_item.content = str(info['tool_calls'][-1].result)
 
-        return ChatAgentResponse(
+        chat_agent_response = ChatAgentResponse(
             msgs=output_messages, terminated=self.terminated, info=info
         )
 
+        # If the output result is a single piece of information, it will be
+        # automatically added to the memory. If it is multiple pieces of
+        # information, the critic agent will need to filter out the results
+        # and then the user will manually add them to the memory to prevent
+        # multiple pieces of information from interfering with LLM.
+        if len(output_messages) == 1:
```

Comment:
```suggestion
        if len(chat_agent_response.msgs) == 1:
```
Similarly for async.